mergify_engine/queue/merge_train.py

```diff
@@ -1574,11 +1635,14 @@ async def _slice_cars(
 
     def find_embarked_pull(
         self, pull_number: github_types.GitHubPullRequestNumber
-    ) -> typing.Optional[EmbarkedPullWithCar]:
-        return first.first(
-            self._iter_embarked_pulls(),
-            key=lambda c: c.embarked_pull.user_pull_request_number == pull_number,
-        )
+    ) -> typing.Union[
+        typing.Tuple[int, EmbarkedPullWithCar],
+        typing.Tuple[typing.Literal[None], typing.Literal[None]],
+    ]:
+        for position, c in enumerate(self._iter_embarked_pulls()):
```

Comment:
Isn't there a better name than just `c` ? Would be better for understanding the code more easily